### PR TITLE
type mapLeft

### PR DESCRIPTION
**Pull request by edgarjrg**
I typed mapLeft as following bimap
```ts
bimap<A, B>(p: Fn<A, B>): <C, D>(q: Fn<C, D>) => (r: Bifunctor<A, C>) => Bifunctor<B, D>;
```

But i'm not sure if i should use

```ts
mapLeft<A, B>(p: Fn<A, B>): <C>(r: Bifunctor<A, C>) => Bifunctor<B, C>;
```

### CODE REVIEW

**Comment by edgarjrg** [comment]
Maybe this is a bit late, of not worth it, but looking to the typings of [Fluture🦋](https://github.com/fluture-js/Fluture), in the case of bifunctors they use 
```ts
export function mapRej<LA, LB, R>(mapper: (reason: LA) => LB): (source: FutureInstance<LA, R>) => FutureInstance<LB, R>

```

I really like this concept of `L` and `R`

May look the 

```ts
mapLeft<LA, LB, R>(p: Fn<LA, LB>): (r: Bifunctor<LA, R>) => Bifunctor<LB, R>;
```

**Comment by davidchambers** [comment]
I don't mind which letters are used for type variables. If you like `LA`, `LB`, and `R`, @edgarjrg, let's use those.

File: types/sanctuary/index.d.ts [hunk 1]
```diff
@@ -171,7 +171,7 @@ declare namespace Sanctuary {
     };
     flip(tk: TK): (tk: TK) => TK;
     bimap<A, B>(p: Fn<A, B>): <C, D>(q: Fn<C, D>) => (r: Bifunctor<A, C>) => Bifunctor<B, D>;
-    mapLeft(tk: TK): (tk: TK) => TK;
+    mapLeft<A, B, C>(p: Fn<A, B>): (r: Bifunctor<A, C>) => Bifunctor<B, C>;
```
Comment (davidchambers) [inline types/sanctuary/index.d.ts]: It won't make a difference in this case—TypeScript will infer `A` = :woman_shrugging:, `B` = :woman_shrugging:, `C` = :woman_shrugging:—but I have seen better results when type variables are introduced as late as possible.

```suggestion
    mapLeft<A, B>(p: Fn<A, B>): <C>(q: Bifunctor<A, C>) => Bifunctor<B, C>;
```

**Comment by vicrac** [comment]
@davidchambers @edgarjrg I'd like to go this way - I mean submitting PRs to this repo, in order not to get the original PR as messy as it starts to be. If you have any unfinished work, please push it now, I will then restrict master write permission.

File: types/sanctuary/index.d.ts (hunk 1 above)
Comment (edgarjrg) [reply types/sanctuary/index.d.ts]: That's what i thought as well 👍

**Comment by edgarjrg** [comment]
@vicrac do you mean like [IOrd](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/37879/files#diff-f860deffe04d8bf92fa45a9aab492619R82) ? to add 
```ts
interface Bifunctor<A, C> extends Functor<C> {
     'fantasy-land/bimap'(..)
}
```

File: types/sanctuary/index.d.ts [hunk 2]
```diff
@@ -171,7 +171,7 @@ declare namespace Sanctuary {
     };
     flip(tk: TK): (tk: TK) => TK;
     bimap<A, B>(p: Fn<A, B>): <C, D>(q: Fn<C, D>) => (r: Bifunctor<A, C>) => Bifunctor<B, D>;
-    mapLeft(tk: TK): (tk: TK) => TK;
+    mapLeft<LA, LB>(p: Fn<LA, LB>): <R>(r: Bifunctor<LA, R>) => Bifunctor<LB, R>;
```
Comment (davidchambers) [inline types/sanctuary/index.d.ts]:
```suggestion
    mapLeft<LA, LB>(p: Fn<LA, LB>): <R>(q: Bifunctor<LA, R>) => Bifunctor<LB, R>;
```

**Comment by vicrac** [comment]
@edgarjrg Yes, I mean like `IOrd`. This allows using custom types with these functions, if only they implement particular `fantasy-land` interface, which I believe (@davidchambers ?) is consistent with `Sanctuary`'s semantics and implementation.

**Comment by davidchambers** [comment]
That is correct, Michał.

**Comment by edgarjrg** [comment]
I added `bimap` in `Bifunctor` sorry for the late commit 🙏

**Comment by vicrac** [comment]
@davidchambers I have no experience with `Bimap` and `Bifunctor` types, could you take a look and tell whether current solution is good enough (I am not sure if primitive types a concern in here)?

**Pull request merged**